### ISSUE TITLE
Extension methods calling other extension methods in same file fail to compile

### ISSUE THREAD

**Issue by johnkslg** [handzlikchris/FastScriptReload#139]
If there are extension methods in the code being recompiled, it errors out with this:

error CS0121: The call is ambiguous between the following methods or properties: 'Eco.Client.Assets.Scripts.Animals.Behaviors.AIExtensions__Patched_.PlayerXZDistance(Eco.Client.Animals.Animal)' and 'Eco.Client.Assets.Scripts.Animals.Behaviors.AIExtensions.PlayerXZDistance(Eco.Client.Animals.Animal)'

**Comment by handzlikchris** [comment]
Linked with #117

**Comment by handzlikchris** [comment]
On second thought it may not be. The error in #117 is different.

For the `The call is ambiguous between the following methods or properties` it looks like you've actually changed an extension method. Likely one that has a method which calls other extension method from same file.

It repros with following example
```
    public static void TestExtension(this ExtensionMethodTest test)
    {
        Debug.Log("TestExtension"); 
    }

    public static void TestExtensionCallingOtherExtensionFromSameFile(this ExtensionMethodTest test)
    {
        test.TestExtension(); 
    }
```

This may be simpler to solve than #117. 

@johnkslg btw - let me know if that's roughly what you're changing.

**Comment by handzlikchris** [comment]
should be fixed with #140
